File: programme/validate-palindrome/validate-palindrome.js
// Write a programme to Validate Palindrom string

function palindrome(str) {
  var re = /[\W_]/g;

  var lowRegStr = str.toLowerCase().replace(re, "");

  var reverseStr = lowRegStr.split("").reverse().join("");

  return reverseStr === lowRegStr;
}

const str1 = "level";
console.log("Input  : " + str1);
console.log("Output : " + palindrome(str1));

console.log("\n");

const str2 = "coding";
console.log("Input  : " + str2);
console.log("Output : " + palindrome(str2));

File: programme/validate-palindrome/validate-palindrome.java
class Main {
    public static void main(String[] args) {
        String str1 = "level";
        System.out.println("Input: " + str1);
        System.out.println("Output: " + isPalindrome(str1));

        System.out.println();

        String str2 = "coding";
        System.out.println("Input: " + str2);
        System.out.println("Output: " + isPalindrome(str2));
    }

    private static boolean isPalindrome(String s) {
        String lowString = s.toLowerCase();

        String reverseString = new StringBuilder(lowString).reverse().toString();

        return lowString.equals(reverseString);
    }
}
